lib/trento/software_updates/discovery.ex

```diff
@@ -254,6 +261,35 @@ defmodule Trento.SoftwareUpdates.Discovery do
 
   defp compute_software_updates_discovery_health(_), do: SoftwareUpdatesHealth.warning()
 
+  defp handle_discovery_result(nil), do: {:error, :not_found}
+
+  defp handle_discovery_result(%DiscoveryResult{failure_reason: failure_reason})
+       when not is_nil(failure_reason),
+       do: {:error, failure_reason_to_atom(failure_reason)}
```

Comment:
I'm not clear why we're squashing errors this way. At this point we could just return whatever result we have and just have the frontend recognizing if `failure_reason` is not nil.

I suppose this was done because otherwise we'd have to rework the frontend? If this was the purpose, I'm fine with it. Maybe we can revisit it in a second time.

Reply:
I wanted to keep the endpoint unchanged in its input/output. Not changing the frontend is indeed a consequence.